### PR TITLE
Diffusers/thuchk

### PR DESCRIPTION
**Pull request by thuc-moreh**
Add train scripts + model_batchsize_file for the tasks: unconditional_image_generation, controlnet, text_to_image
Add setup.sh (create env and intsall reqs), run_task.sh (for training 1 model or all model in model_batchsize_file for each task)
Other files for checking and recording GPU memory according to Huong.

### CODE REVIEW

**Review by kimmorehvn — APPROVED** [review]
good works

File: examples/controlnet/train.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Activate conda env
+base_env=$(conda info | grep -i 'base environment' | awk -F': ' '{print $2}' | sed 's/ (read only)//' | tr -d ' ')
+source ${base_env}/etc/profile.d/conda.sh
+conda activate ${env_name}
+moreh-switch-model -M 1 
+
+# Run training script
+echo "# ========================================================= #"
+echo "training ${model_name}.."
+python train_controlnet.py \
```
Comment (loctxmoreh) [inline examples/controlnet/train.sh]: Activating `conda` environment in a shell script is (sort of) a hassle. I suggest running `train_controlnet.py` by using `conda run`. For example: `conda run -n ${env_name} python3 train_controlnet.py ...`

File: examples/text_to_image/train.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+# Activate conda env
+base_env=$(conda info | grep -i 'base environment' | awk -F': ' '{print $2}' | sed 's/ (read only)//' | tr -d ' ')
+source ${base_env}/etc/profile.d/conda.sh
+conda activate ${env_name}
+
+# Run training script
+echo "# ========================================================= #"
+echo "training ${model_name}.."
+python train_text_to_image.py \
```
Comment (loctxmoreh) [inline examples/text_to_image/train.sh]: Same case. `conda run -n $env_name python3 train_text_to_image.py ...` looks cleaner

File: examples/unconditional_image_generation/train.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+# Activate conda env
+base_env=$(conda info | grep -i 'base environment' | awk -F': ' '{print $2}' | sed 's/ (read only)//' | tr -d ' ')
+source ${base_env}/etc/profile.d/conda.sh
+conda activate ${env_name}
+
+# Run training script
+echo "# ========================================================= #"
+echo "training ${model_name}.."
+python train_unconditional.py \
```
Comment (loctxmoreh) [inline examples/unconditional_image_generation/train.sh]: `conda run -n $env_name python3 train_unconditional.py ...` seems better

File: examples/unconditional_image_generation/train_unconditional.py [hunk 4]
```diff
@@ -467,7 +475,7 @@ def load_model_hook(models, input_dir):
     # Preprocessing the datasets and DataLoaders creation.
     augmentations = transforms.Compose(
         [
-            transforms.Resize(args.resolution, interpolation=transforms.InterpolationMode.BILINEAR),
+            transforms.Resize(args.resolution, interpolation=PIL.Image.BICUBIC),
```
Comment (loctxmoreh) [inline examples/unconditional_image_generation/train_unconditional.py]: I believe this modification is only needed for `torch<=1.7`. I reverted this change and the script run fine.

Not to mention bilinear and bicubic are two different interpolation modes...

File: examples/unconditional_image_generation/train_unconditional.py [hunk 5]
```diff
@@ -617,51 +627,53 @@ def transform_images(examples):
                 logs["ema_decay"] = ema_model.cur_decay_value
             progress_bar.set_postfix(**logs)
             accelerator.log(logs, step=global_step)
+        
+        loss_dict["loss_epoch_"+str(epoch)]= loss.detach().item()
```
Comment (loctxmoreh) [inline examples/unconditional_image_generation/train_unconditional.py:631]: `loss` is already recorded before (in line 615). Why don't we reuse that instead?

File: examples/unconditional_image_generation/train_unconditional.py (hunk 4 above)
Comment (thuc-moreh) [reply examples/unconditional_image_generation/train_unconditional.py]: Thanks for letting me know! I've updated the code to reverted back to the old interpolation mode.

File: examples/unconditional_image_generation/train_unconditional.py (hunk 5 above)
Comment (thuc-moreh) [reply examples/unconditional_image_generation/train_unconditional.py:631]: Yes, I'm using the loss calculated earlier in the code, and I'm creating a dictionary to save the loss for each epoch.
